src/MicrosoftTeams.ts

```diff
@@ -745,12 +770,20 @@ namespace microsoftTeams {
          * Notifies the frame that initiated this authentication request that the request was successful.
          * This function is usable only on the authentication window.
          * This call causes the authentication window to be closed.
-         * @param result Specifies a result for the authentication. If specified, the frame that initiated the authentication pop-up receives this value in its callback.
+         * @param authenticationResultParams Specifies a result for the authentication. It contains the optional values of state(URL to redirect back after authentication in Outlook Desktop), reason(in case of failure), result(in case of success).
          */
-        export function notifySuccess(result?: string): void {
+        export function notifySuccess(authenticationResultParams: AuthenticationResultParameters): void {
+            if (authenticationResultParams.state) {
+                let decodedUrl = decodeURIComponent(authenticationResultParams.state);
+                if (isValidState(decodedUrl)) {
+                    let link = document.createElement("a");
+                    link.href = updateUrlParameter(decodedUrl, "result", authenticationResultParams.result);
```

Comment:
This is dangerous because the result might be a token and this code would be putting it in the QSPs which are generally not secure (even on an https URL). When would a caller provide both a state and a result? In those cases what does the result typically contain?

Reply:
state should be passed when the client is Outlook desktop, result depends on the 3rd party if they want any information back in their successCallback method they can choose to pass it in  result.
we can either
1. Document that it would be part of QSP so they should not assign sensitive information to it.
2. We can remove support for result altogether for connectors but we need to make sure the solution is consistent for all clients.